### ISSUE TITLE
Questions about the sampling strategy in the paper.

### ISSUE THREAD

**Issue by AlexsaseXie** [FuxiCV/Ladybird#1]
I just cannot understand the sampling strategy part of this paper. I have several questions as follows:

1. In section 1.1, I cannot understand the what is the Fourier power spectra. Furthurmore, what's the relationship between the difference in Fourier power spectra shown in Fig.3 and your following discussions? 

> "Blue noise samplers [25] are well-known to show good improvements for low-dimensional integration problems whereas low-discrepancy [22] samplers like Halton [12] and Sobol [13] are more effective for higher dimensional problems. In this work, we use farthest point selection strategy [7] from any given pointset to select our samples."

2. I understand that a lower star discrepancy of your sampled points will produce of a lower expectation upper bound of the error in the quasi-monte carlo integration(in the red box).
![image](https://user-images.githubusercontent.com/26740167/110776695-530f8580-829b-11eb-9f30-21c74246f43e.png)
But the loss in training the sdf estimation network is the L-2 loss between gt and the predicted sdf on all sample points as formulated in your paper:
![image](https://user-images.githubusercontent.com/26740167/110777005-9a961180-829b-11eb-899d-3c07117b1e4d.png)

**So what's the relationship between a better quasi-monte carlo integration above(in the red box) and a better final neural network g_w prediction?** There is a square so that you cannot separate the integration. Lower error above cannot result in a lower error below when you replace g_w(I, p) with X_i (under the assumption that the neural network produces accurate predictions). I cannot find any explanation in your paper. 

By the way, I think a good evaluation of the g_w should be the error defined with m(p) removed as follows:
![image](https://user-images.githubusercontent.com/26740167/110777391-06787a00-829c-11eb-96b0-0ed08229b723.png)
where the integration is calculated over R^d with a uniform distribution. But still I cannot reason the relationship. 

I'm looking forward to your reply.
Also, are there codes to upload?

**Comment by Uviveknarayan** [comment]
Did you get the code from the authors? I too require the code for a personal side project.
